customer-bill/src/main/java/org/fiware/tmforum/customer_bill/rest/CustomerBillOnDemandApiController.java

```diff
@@ -0,0 +1,91 @@
+package org.fiware.tmforum.customer_bill.rest;
+
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.annotation.Controller;
+import io.reactivex.Single;
+import io.reactivex.schedulers.Schedulers;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.fiware.customer_bill.api.CustomerBillOnDemandApi;
+import org.fiware.customer_bill.model.CustomerBillOnDemandCreateVO;
+import org.fiware.customer_bill.model.CustomerBillOnDemandVO;
+import org.fiware.tmforum.common.exception.NonExistentReferenceException;
+import org.fiware.tmforum.common.validation.ReferenceValidationService;
+import org.fiware.tmforum.customer_bill.TMForumMapper;
+import org.fiware.tmforum.customer_bill.domain.customer_bill.CustomerBillOnDemand;
+import org.fiware.tmforum.customer_bill.exception.CustomerBillOnDemandCreationException;
+import org.fiware.tmforum.customer_bill.repository.CustomerBillOnDemandRepository;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+@Slf4j
+@Controller
+@RequiredArgsConstructor
+public class CustomerBillOnDemandApiController implements CustomerBillOnDemandApi {
+
+    private final TMForumMapper tmForumMapper;
+    private final CustomerBillOnDemandRepository customerBillOnDemandRepository;
+    private final ReferenceValidationService validationService;
+
+
+    @Override
+    public Single<HttpResponse<CustomerBillOnDemandVO>> createCustomerBillOnDemand(CustomerBillOnDemandCreateVO customerBillOnDemandCreateVO) {
+
+        CustomerBillOnDemandVO customerBillOnDemandVO = tmForumMapper.map(customerBillOnDemandCreateVO);
+        CustomerBillOnDemand customerBillOnDemand = tmForumMapper.map(customerBillOnDemandVO);
+
+        Single<CustomerBillOnDemand> customerBillOnDemandSingle = Single.just(customerBillOnDemand);
+        Single<CustomerBillOnDemand> checkingSingle;
+
+        /**
+         * Validate references
+         *
```

Comment:
Question to be solved is, if the references are allowed to exist outside the current system(probably), thus we need to check referential integrity within the broker and potential external implementations of the TMForum API